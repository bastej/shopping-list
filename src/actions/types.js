export const CREATE_NEW_CART = "CREATE_NEW_CART";
export const DELETE_CART = "DELETE_CART";
export const ADD_CART_PRODUCT = "ADD_PRODUCT";
export const DELETE_CART_PRODUCT = "DELETE_CART_PRODUCT";
export const UPDATE_CART_PRODUCT_COUNT = "UPDATE_CART_PRODUCT_COUNT";

export const ADD_CATEGORY = "ADD_CATEGORY";
export const SET_NAV_HEADER = "SET_NAV_HEADER";
export const CALCULATE_NUTRIENTS = "CALCULATE_NUTRIENTS";
export const GET_PRODUCT_HINTS = "GET_PRODUCT_HINTS";
export const CLEAR_PRODUCT_HINTS = "CLEAR_PRODUCT_HINTS";

export const CREATE_NEW_MEAL = "CREATE_NEW_MEAL";
export const DELETE_MEAL = "DELETE_MEAL";
export const ADD_MEAL_PRODUCT = "ADD_MEAL_PRODUCT";
export const DELETE_MEAL_PRODUCT = "DELETE_MEAL_PRODUCT";
export const UPDATE_MEAL_PRODUCT_COUNT = "UPDATE_MEAL_PRODUCT_COUNT";

export const SET_NEXT_MONTH = "SET_NEXT_MONTH";
export const SET_PREV_MONTH = "SET_PREV_MONTH";
export const SELECT_START_DATE = "SELECT_START_DATE";
export const SELECT_END_DATE = "SELECT_END_DATE";
export const CREATE_DIET_PLAN = "CREATE_DIET_PLAN";
