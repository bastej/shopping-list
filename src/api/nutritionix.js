import axios from "axios";

const API_KEY = "66dc95ac590ef97c7de66e82397a3853";
const APP_KEY = "6bbdffac";

export const nutritionixHints = axios.create({
  baseURL: "https://trackapi.nutritionix.com/v2",
  headers: {
    "x-app-id": APP_KEY,
    "x-app-key": API_KEY,
    "x-remote-user-id": 0
  }
});
