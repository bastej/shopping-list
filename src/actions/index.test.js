import { createNewCart, updateProductCount, addProduct } from "./index.js";
import { UPDATE_CART_PRODUCT_COUNT } from "./types";
import axios from "axios";
import { nutritionixAPI } from "../api/nutritionix";

describe("cart actions", () => {
  describe("updateProductCount", () => {
    it("should return cart action", () => {
      expect(updateProductCount(0, 0, "", "cart")).toEqual({
        type: UPDATE_CART_PRODUCT_COUNT,
        payload: {
          actionType: "",
          listID: 0,
          productID: 0
        }
      });
    });
  });
  describe("API testing", () => {
    it("should fetch from api info about 4 specific product nutrients", async () => {
      const mockedName = "banana";
      expect.assertions(1);
      const data = await nutritionixAPI.get("/search/instant/", {
        params: {
          common_general: true,
          detailed: true,
          claims: true,
          branded: false,
          query: mockedName
        }
      });
      expect(data.data.common[0]).toMatchSnapshot({
        full_nutrients: expect.arrayContaining([
          expect.objectContaining({ attr_id: 203 }),
          expect.objectContaining({ attr_id: 204 }),
          expect.objectContaining({ attr_id: 205 }),
          expect.objectContaining({ attr_id: 208 })
          // TODO: throw error with message when uncomment line below
          // expect.objectContaining({ attr_id: 1208 })
        ])
      });
    });
  });
});

// if axios is mocked with below, nutritionixAPI doesn't work
// jest.mock("axios");
// it("action should receive object", async () => {
//   const product = { name: "banana" };
//   const mockedResponse = { data: product };
//   axios.get.mockResolvedValue(mockedResponse);
//   const response = await axios.get();

//   expect(response.data).toEqual(product);
// });
