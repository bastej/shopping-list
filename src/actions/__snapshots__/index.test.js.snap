// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart actions API testing should fetch from api info about 4 specific product nutrients 1`] = `
Object {
  "claims": Array [
    "Cholesterol Free",
    "Low Cholesterol",
    "Fat Free",
    "Low Fat",
    "Good Source of Fiber",
    "Saturated Fat Free",
    "Low Saturated Fat",
    "Sodium Free",
    "Very Low Sodium",
    "Low Sodium",
  ],
  "common_type": null,
  "food_name": "banana",
  "full_nutrients": ArrayContaining [
    ObjectContaining {
      "attr_id": 203,
    },
    ObjectContaining {
      "attr_id": 204,
    },
    ObjectContaining {
      "attr_id": 205,
    },
    ObjectContaining {
      "attr_id": 208,
    },
  ],
  "locale": "en_US",
  "photo": Object {
    "thumb": "https://d2xdmhkmkbyw75.cloudfront.net/399_thumb.jpg",
  },
  "serving_qty": 1,
  "serving_unit": "medium (7\\" to 7-7/8\\" long)",
  "serving_weight_grams": 118,
  "tag_id": "399",
  "tag_name": "banana",
}
`;
